### PR TITLE
Fix Wallos Update

### PR DESCRIPTION
**Pull request by MickLesk**
## Description
Check if wallos.db exists
Minor optimization in the backup of files

Fixes # (issue)
https://github.com/community-scripts/ProxmoxVE/issues/315#issuecomment-2483634383

## Type of change
Please check the relevant option(s):

- [x] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Testing performed (I have tested my changes, ensuring everything works as expected)
- [x] Documentation updated (I have updated any relevant documentation)

### CODE REVIEW

**Comment by MickLesk** [comment]
There is a small additional bugfix included, namely that all files + subfolders (/*) are also backed up. 

About the DB:
I have extra pulled 3 different LXC, even on Ubuntu. Each time I had the file (but only after registering). If someone updates before registering, the bugfix will take effect.

File: ct/wallos.sh [hunk 1]
```diff
@@ -63,14 +63,19 @@ if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_v
   msg_info "Updating ${APP} to ${RELEASE}"
   cd /opt
   wget -q "https://github.com/ellite/Wallos/archive/refs/tags/v${RELEASE}.zip"
-  mv /opt/wallos/db/wallos.db /opt/wallos.db
-  mv /opt/wallos/images/uploads /opt/images/uploads
+  mkdir -p /opt/uploads
+  if [ -f "/opt/wallos/db/wallos.db" ]; then
+  mv "/opt/wallos/db/wallos.db" "/opt/wallos.db"
+  fi
+  mv /opt/wallos/images/uploads/* /opt/uploads_backup/
```
Comment (burgerga) [inline ct/wallos.sh:70]: Should be /opt/uploads

Comment (MickLesk) [reply ct/wallos.sh:70]: For what? Its only a Backup for 4 lines of Code. Moved Back later

Comment (burgerga) [reply ct/wallos.sh:70]: Your mkdir creates a different folder than the one you're trying to use

File: ct/wallos.sh [hunk 2]
```diff
@@ -63,14 +63,19 @@ if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_v
...
   unzip -q v${RELEASE}.zip
   rm -rf /opt/wallos
   mv Wallos-${RELEASE} /opt/wallos
   rm -rf /opt/wallos/db/wallos.empty.db
-  mv /opt/wallos.db /opt/wallos/db/wallos.db
-  mv /opt/images/uploads /opt/wallos/images/uploads
+  if [ -f "/opt/wallos.db" ]; then
+  mv "/opt/wallos.db" "/opt/wallos/db/wallos.db"
+  fi
+  mv /opt/uploads_backup/* /opt/wallos/images/uploads
```
Comment (burgerga) [inline ct/wallos.sh:78]: `mv: cannot move '/opt/uploads_backup/icons' to '/opt/wallos/images/uploads/icons': Directory not empty`
It's a bit ugly by the original author, but uploads/icons contain the default icons :(

Comment (MickLesk) [reply ct/wallos.sh:78]: Then only we delete the folder before

**Comment by havardthom** [comment]
> There is a small additional bugfix included, namely that all files + subfolders (/*) are also backed up.
> 
> About the DB: I have extra pulled 3 different LXC, even on Ubuntu. Each time I had the file (but only after registering). If someone updates before registering, the bugfix will take effect.

In `wallos-install.sh` we do `mv /opt/wallos/db/wallos.empty.db /opt/wallos/db/wallos.db` so it should always exist? I don't think we need all these if checks

File: ct/wallos.sh (hunk 2 above)
Comment (burgerga) [reply ct/wallos.sh:78]: I did a code search in https://github.com/ellite/Wallos, an if I understand correctly ONLY /opt/wallos/images/uploads/logos/ is used for user-uploaded images. (which is probably also why only this directory and the db directory are mentioned as docker volumes in the [docker instructions](https://github.com/ellite/Wallos?tab=readme-ov-file#docker-1))

Comment (MickLesk) [reply ct/wallos.sh]: so switched to logo. Can you Test?

**Comment by MickLesk** [comment]
> > There is a small additional bugfix included, namely that all files + subfolders (/*) are also backed up.
> > About the DB: I have extra pulled 3 different LXC, even on Ubuntu. Each time I had the file (but only after registering). If someone updates before registering, the bugfix will take effect.
> 
> In `wallos-install.sh` we do `mv /opt/wallos/db/wallos.empty.db /opt/wallos/db/wallos.db` so it should always exist? I don't think we need all these if checks



Removed for the moment. Let's see if there are problems like with the one user. However that could be.

**Comment by havardthom** [comment]
Only logos needs to be backed up now? Tested that everything works?

**Review by havardthom — APPROVED** [review]

**Comment by burgerga** [comment]
Tested and seems to work, not that it counts as official review, but wanted to let you know ;)

**Pull request merged**